brave/brave-ios
# Fix #2786: Remove yahoo se for unsupported regions.

This commit also refactors logic of selecting initial search engines.



## Summary of Changes


This pull request fixes #2786 

## Submitter Checklist:


## Test Plan:

New install test
- Install new build with Polish locale
- verify that yahoo isn't on the list

Upgrade test
- Install 1.19.2/1.20 with Polish locale
- Set yahoo as your default search
- Upgrade to 1.21
- Verify that yahoo is not showing anymore
- For Polish locale the default show now be Google I think

I refactor the search engine logic a little so please test this locales as well to catch regressions.

New install, other regions:
- `US`, see if yahoo is present and showing first on the list, google should be default
- `DE`, see if yahoo is present and showing first on the list, ddg should be default, yahoo url should be `de.yahoo.com..`
- `FR`, see if yahoo is present and showing first on the list, qwant should be default, yahoo url should be `fr.yahoo.com..`


## Screenshots:



## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
